pkg/console/controllers/oidcsetup/oidcsetup.go

```
@@ -252,7 +256,14 @@ func (c *oidcSetupController) checkClientConfigStatus(authnConfig *configv1.Auth
 		return false, "deployment unavailable or outdated", nil
 	}
 
-	if clientSecret.GetResourceVersion() != depl.ObjectMeta.Annotations["console.openshift.io/oauth-secret-version"] {
+	// Get the TARGET secret (synced copy in openshift-console namespace)
+	// to compare its resource version with the deployment annotation
+	targetClientSecret, err := c.targetNSSecretsLister.Secrets(api.OpenShiftConsoleNamespace).Get("console-oauth-config")
+	if err != nil {
+		return false, "", err
+	}
+
+	if targetClientSecret.GetResourceVersion() != depl.ObjectMeta.Annotations["console.openshift.io/oauth-secret-version"] {
```

golangci-lint (2.12.2)

[error] QF1008: could remove embedded field "ObjectMeta" from selector

(staticcheck)